# Distinct Gut and Oral Microbiome Patterns Associated with Dyslexia in a Family-Based Cohort
This study explored whether individuals with dyslexia or childhood apraxia of speech (CAS) exhibit distinct gut and oral microbiome signatures from their neurotypical family members. 

## Project Overview
This repository contains the scripts, data, and documentation for this study. Dyslexia and apraxia are neurodevelopmental disorders affecting language processing and motor coordination, respectively. While traditionally studied from neurological and genetic perspectives, emerging research suggests that the gut-brain and oral-brain axes may play a crucial role in these conditions. 

This project characterized the microbial communities in the gut and oral microbiome to explore their potential associations with dyslexia and CAS. Using Illumina 16S rRNA sequencing (i.e., V4 amplicon region) data, we characterize the microbial diversity, composition, and functional potential of individuals with dyslexia and apraxia. Sequencing data was deposited in the NCBI SRA website with the accession ID: PRJNA1246995. 

### Objectives 

- Characterize the microbial community of individuals with dyslexia/apraxia and their neurotypical family members.
- Investigate the presence of specific microbial taxa and metabolic pathways that may be linked to these phenotypes.

### Repository Structure 
- 01_Scripts
- 02_Results
- 03_Docs

### Required software 
- QIIME2 (v.2024.5)
- R (v.4.1)
- Python (v.3.2)
- Unix

### Taxonomic Databases 
- Greengenes2 (v.2024.09; full-length 16S database)
- SILVA (v.138, 99%)
- MIMt2.0
- GSR-DB (full-length 16S database)

#### Citation
If you use this repository or its resources in your research, please cite: XYZ

#### Acknowledgements 
All procedures involving human participants were reviewed and approved by the Institutional Review Board at Arizona State University (IRB #24244 “Biology of Language”) in accordance with the ethical standards of the World Medical Association. Initial enrollment and data collection were conducted at the University of Washington (IRB #34416). Written informed consent was obtained from all adult participants, and parental consent was provided for all minor participants. The microbiome analysis was supported by a seed grant awarded to B.P. from the Institute for Social Science Research (ISSR) at Arizona State University. We also thank the participants and their families for their time and commitment to this research. We are also grateful to the staff at the ASU iLab for their technical support and assistance with sequencing. This work utilized computational resources provided by the Sol Supercomputer at Arizona State University.

#### Author Contributions:
S.L.W. and B.P. contributed to the study’s conceptualization and overall research design. S.L.W., M.J., Y.K., M.O., and A.H. performed experiments and data analysis. S.L.W., M.J., M.O., and A.H. processed and analyzed the 16S rRNA sequences. Y.K. conducted clinical assessments. S.L.W. wrote the initial draft of the manuscript. B.P., C.M.W., M.J., Y.K., M.O., and A.H. contributed to manuscript revisions. All authors reviewed and approved the final submitted version.

